homeassistant/components/smartthings/lock.py

```diff
@@ -0,0 +1,54 @@
+"""
+Support for locks through the SmartThings cloud API.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/smartthings.lock/
+"""
+from homeassistant.components.lock import LockDevice
+from homeassistant.const import STATE_LOCKED
+
+from . import SmartThingsEntity
+from .const import DATA_BROKERS, DOMAIN
+
+DEPENDENCIES = ['smartthings']
+
+
+async def async_setup_platform(hass, config, async_add_entities,
+                               discovery_info=None):
+    """Platform uses config entry setup."""
+    pass
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Add locks for a config entry."""
+    broker = hass.data[DOMAIN][DATA_BROKERS][config_entry.entry_id]
+    async_add_entities(
+        [SmartThingsLock(device) for device in broker.devices.values()
+         if is_lock(device)])
+
+
+def is_lock(device):
+    """Determine if the device supports the lock capability."""
+    from pysmartthings import Capability
+    if Capability.lock in device.capabilities:
+        return True
+    return False
```

Comment:
```python
return Capability.lock in device.capabilities
```
?

Reply:
There is also a `lockOnly` capability which is considered deprecated, but is likely still in use.  We need to review the existing Device Type Handlers from the SmartThings Public github repo to determine if this capability should be supported.

Reply:
Good catch - was going to hunt down some of those other capabilities but forgot. Will look and see what is out there and what is getting used, don't think its worth committing too much effort supporting additional/deprecated capabilities if they aren't really getting used.

Reply:
let’s skip it for now as pysmarttings doesn’t have it in the inventory therefore the integration won’t subscribe to the capability.  I’ll do some digging tonight to see if it’s in use.  Some legacy capabilities are super prevalent like thermostat which nearly all handlers use, instead of the newer representations.

Reply:
So here's the good news.  I reviewed all DTHs and couldn't find any that use `lockOnly`.  I did find that the zigbee/zwave handlers also implement an undocumented capability called `lockCodes`.  This isn't on the old or new capabilities reference, so suggest we ignore it.